marimo/_plugins/ui/_impl/table.py

```diff
@@ -693,6 +703,12 @@ def _get_column_summaries(self, args: EmptyArgs) -> ColumnSummaries:
             is_disabled=False,
         )
 
+    def _get_data_url(self, args: EmptyArgs) -> GetDataUrlResponse:
+        """Get the data URL for the entire table. Used for charting."""
+        del args
+        # TODO: May want to filter data here
+        return GetDataUrlResponse(data_url=self._searched_manager.to_data({}))
```

Comment:
for charts for performance, you may want to use `_to_marimo_arrow()` which first tries to convert the data to arrow format and then to csv. which will help with performance (less over the wire and less serialization/deserialization